corehq/apps/locations/models.py

```diff
@@ -399,35 +397,58 @@ class SQLLocation(SyncSQLToCouchMixin, MPTTModel):
     active_objects = OnlyUnarchivedLocationManager()
 
     @classmethod
-    def _migration_get_fields(cls):
+    def get_sync_fields(cls):
         return ["domain", "name", "site_code", "external_id",
                 "metadata", "is_archived"]
 
-    @classmethod
-    def _migration_get_couch_model_class(cls):
-        return Location
+    def sync_couch_location(self):
+        if self.location_id:
+            couch_obj = self.couch_location
+        else:
+            couch_obj = Location()
+            couch_obj._id = uuid.uuid4().hex
+            self.location_id = couch_obj._id
```

Comment:
We have a lot of instances where we save something to the db to get an ID, even though it would be simpler to save later.  We should really get more comfortable setting the ID during creation when it makes sense to do so.